pkg/storage/backend-storage/backend-storage_test.go

```diff
@@ -0,0 +1,166 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+	virtv1 "kubevirt.io/api/core/v1"
+	cdifake "kubevirt.io/client-go/generated/containerized-data-importer/clientset/versioned/fake"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	storagetypes "kubevirt.io/kubevirt/pkg/storage/types"
+	"kubevirt.io/kubevirt/pkg/testutils"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+var _ = Describe("Backend Storage", func() {
+	var backendStorage *BackendStorage
+	var cdiClient *cdifake.Clientset
+	var config *virtconfig.ClusterConfig
+	var kvInformer cache.SharedIndexInformer
+	var storageClassInformer cache.SharedIndexInformer
+
+	cdiConfigInit := func() (cdiConfig *v1beta1.CDIConfig) {
+		cdiConfig = &v1beta1.CDIConfig{
+			ObjectMeta: k8smetav1.ObjectMeta{
+				Name: storagetypes.ConfigName,
+			},
+			Spec: v1beta1.CDIConfigSpec{
+				UploadProxyURLOverride: nil,
+			},
+			Status: v1beta1.CDIConfigStatus{
+				FilesystemOverhead: &v1beta1.FilesystemOverhead{
+					Global: storagetypes.DefaultFSOverhead,
+				},
+			},
+		}
+		return
+	}
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient := kubecli.NewMockKubevirtClient(ctrl)
+		kubevirtFakeConfig := &virtv1.KubeVirtConfiguration{}
+		config, _, kvInformer = testutils.NewFakeClusterConfigUsingKVConfig(kubevirtFakeConfig)
+		storageClassInformer, _ = testutils.NewFakeInformerFor(&storagev1.StorageClass{})
+		cdiConfig := cdiConfigInit()
+		cdiClient = cdifake.NewSimpleClientset(cdiConfig)
+		virtClient.EXPECT().CdiClient().Return(cdiClient).AnyTimes()
+
+		backendStorage = NewBackendStorage(virtClient, config, storageClassInformer.GetStore())
+	})
+
+	Context("Storage class", func() {
+		It("Should return VMStateStorageClass when set", func() {
+			kvCR := testutils.GetFakeKubeVirtClusterConfig(kvInformer)
+			kvCR.Spec.Configuration.VMStateStorageClass = "myfave"
+			testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, kvCR)
+			sc, err := backendStorage.getStorageClass()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sc).To(Equal("myfave"))
+		})
+
+		It("Should return the default storage class when VMStateStorageClass is not set", func() {
+			for i := 0; i < 5; i++ {
+				sc := storagev1.StorageClass{
+					ObjectMeta: k8smetav1.ObjectMeta{
+						Name: fmt.Sprintf("sc%d", i),
+					},
+				}
+				if i == 3 {
+					sc.Annotations = map[string]string{"storageclass.kubernetes.io/is-default-class": "true"}
+				}
+				err := storageClassInformer.GetStore().Add(&sc)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			sc, err := backendStorage.getStorageClass()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sc).To(Equal("sc3"))
+		})
+	})
+
+	Context("Access mode", func() {
+		BeforeEach(func() {
+			cdiClient.PrependReactor("get", "storageprofiles", func(action testing.Action) (bool, runtime.Object, error) {
```

Comment:
Actually we don't need to prepend reactor.
We can store the storageProfiles with `Create` and let the fakeclient to be used as a real client.
IMHO it's easier to understand and more robust

Reply:
Awesome, looks much better, pushed, PTAL!